source/Octopus.Client/Repositories/Async/MixedScopeBaseRepository.cs

```diff
@@ -104,5 +102,21 @@ protected void EnsureSingleSpaceContext()
                     }, () => throw new SingleSpaceOperationInMultiSpaceContextException());
                 });
         }
+
+        protected void EnsureSystemContext()
+        {
+            var exception = new Exception("Attempted to perform a system operation in a space scoped context. Ensure you are in system context first by using client.ForSystem()");
```

Comment:
Can we create a more specific exception type here? Better to use specific types for libraries

Reply:
Yeah I guess you're right, people are unlikely to programatically handle any custom exception here. Can we at least use a more specific built-in exception type? eg `InvalidOperationException`?

Reply:
fixed